### DISCUSSION TITLE
How to assign value?

### DISCUSSION THREAD

**Discussion by amosaha** [idaholab/moose#20416] [Q&A General] [answered]
Hello，
I have a set of 1D data along the z axis and then need to interpolate to the 2D boundaries of a 3D grid, how can I achieve this?

**Comment by GiudGiud** [comment, marked as answer]
hello

Is this to initialize a variable? I ll assume so.
Use a FunctionIC to initialize the variable
https://mooseframework.inl.gov/source/ics/FunctionIC.html
with a PiecewiseConstant function with `axis = z`
https://mooseframework.inl.gov/source/functions/PiecewiseConstant.html

Guillaume
